### DISCUSSION TITLE
Project Meeting 2021/08/04

### DISCUSSION THREAD

**Discussion by moelang** [autocore-ai/SDV#19] [Show and tell]
# 第二阶段工作交流

## 实现基础框架：在[zenoh-flow](https://github.com/autocore-ai/zenoh-flow)中按照如下数据流图建立自动驾驶[规划](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/tree/master/src/planning)规控流程
![](https://raw.githubusercontent.com/autocore-ai/SDV/bffb0c05af988501cd3d2a965685ab02973c77e9/img/nodes.svg)

### 流程解释
* 每个圆框表示需要暴露给`zenoh-flow`作为`operator`的动态库，它们又各自依赖支撑具体功能的多个动态库
* `map_file`地图文件作为外部远程资源加载
* `goal_pose`在`CloudViewer`由用户手动输入
* `lanelet2_global_planner`根据地图文件和目的地位姿、车辆当前位姿一次性计算出全局路径`global_path`
* `behaviour_planner`根据当前场景选择调用`lane_planner`或`parking_planner`，根据当前地图语义以及红绿灯信息规划出`trajectory`
* `object_collision_estimator`结合其他目标信息对轨迹做限速处理，并生成最终轨迹
* `pure_pursuit`控制单元根据车身位姿和轨迹计算出车辆控制数据`command`
* `simple_planning_simulator`执行控制数据并更新车辆位姿`current_pose`

### 工作安排
* 节点逐个开发，预计会到10月中可以联动
* 节点将会包括并不仅限于`Autoware.Auto`的当前功能，所作调整修改较多，届时会形成一个新的独立分支。
* 红绿灯模块需要从其他方案引入

### 可能的工作项（待评估）
* 推理平台(eg: [TensorRT](https://github.com/NVIDIA/TensorRT))的封装与对接`zenoh-flow`
* 对接带图像输出的仿真器例如`LGSVL`
* 需要对应的模型
* 对接需要的例如摄像头驱动、图像前处理、坐标转换等节点
* 需要与ODD同事讨论 @Sharrrrk

**Comment by aicarmark** [comment]
推理平台，可以用TVM（支持GPU/TensorRT后端），有rust binding

**Reply by moelang** [reply]
> 推理平台，可以用TVM（支持GPU/TensorRT后端），有rust binding

是[这个](https://github.com/apache/tvm/tree/main/rust/tvm)吗?
